ui/shared/ConfirmationDialog.qml

```diff
@@ -47,11 +65,17 @@ ModalPopup {
         StatusButton {
             id: confirmButton
             type: confirmationDialog.btnType
-            anchors.right: cancelButton.left
-            anchors.rightMargin: Style.current.smallPadding
+            anchors.right: cancelButton.visible ? cancelButton.left : parent.right
+            anchors.rightMargin: cancelButton.visible ? Style.current.smallPadding : 0
             text: confirmationDialog.confirmButtonLabel
             anchors.bottom: parent.bottom
-            onClicked: confirmationDialog.confirmButtonClicked()
+            onClicked: {
+                if (executeConfirm && typeof executeConfirm === "function") {
```

Comment:
why do I need this `executeConfirm`? as a user I could simply implement `onConfirmButtonClicked` handler.